text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,647 @@
+    Run `cdk-launchpad update` to upgrade it to the newest version.
+```
+
+### Ways of customizing the Launchpad
+
+The default Launchpad contains the same resources as the `cdk bootstrap` command (in version 2) creates.
+A CDK Launchpad can be customized with the following additional options:
+
+1. Compliance rules.
+2. Additional resources (deployed using CloudFormation, CDK, or in any other way, and simply referenced by the Launchpad).
+3. Modifications to the CDK bootstrap resources (using either CloudFormation templates or CDK).
+4. Setting IAM Permission Boundaries used by all applications deployed into a given environment.
+
+There are two ways of customizing a Launchpad: using JSON/YAML files, or using CDK code.
+
+#### Customizing the Launchpad with JSON/YAML files
+
+This option of customizing the Launchpad is geared primarily towards platform administrators
+who don't want to use a programming language (and thus, CDK code) to customize their Launchpads.
+Using a JSON or YAML file in a specific format allows customizing all aspects of the Launchpad,
+without the need to write a single line of code.
+
+The JSON file has the basic structure (the YAML file is very similar):
+
+```json
+{
+  "launchpad": {
+    ...
+  }
+}
+```
+
+We show examples of different parts of the JSON file in each of the sections below.
+
+You apply the changes using the JSON/YAML file by passing the file as an argument to the Launchpad commands,
+for example `cdk-launchpad update my-launchpad.json/my-launchpad.yaml`.
+
+Deploying the Launchpads through a pipeline will also be supported with
+[CDK Pipelines](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html).
+
+#### Customizing the Launchpad with CDK code
+
+It's also possible to customize the Launchpad using CDK code.
+In that scenario, the Launchpad is basically a CDK application, and you execute the Launchpads CLI commands,
+like `cdk-launchpad update`, inside of it, similarly like you execute CDK CLI commands inside your CDK application.
+
+A CDK application for a Launchpad has a very similar structure to a "regular" CDK application,
+but the classes used are slightly different (the example is in TypeScript):
+
+```ts
+import * as lpads from '@aws-cdk/cdk-launchpads';
+
+const launchpad = new lpads.Launchpad();
+new lpads.StandardCdkBootstrap(launchpad, 'CdkBootstrap', {
+  // options to customize the CDK bootstrap can be passed here...
+});
+
+// additional things can be changed in the Launchpad here...
+
+launchpad.synth();
+```
+
+This CDK application will synthesize to the JSON configuration mentioned above,
+similarly like "regular" CDK applications synthesize to CloudFormation templates.
+
+We will show details of customizing the Launchpad using CDK code in each of the sections below.
+
+### Launchpad features
+
+#### Compliance
+
+Compliance rules allow administrators to restrict what kind of AWS resources can be deployed in a given environment.
```

Comment:
Are compliance rules similar to AWS Control Tower [Guardrails](https://docs.aws.amazon.com/controltower/latest/userguide/guardrails.html)?

Guardrails use AWS Config and these seem to use CfnGuard. However, as a Control Tower user, it would be really nice to have a single view into how accounts are governed and whether or not they're in compliance.

Reply:
The main reason for using CFN Guard over Config is that, at least as far as I know, Config rules cannot fail deployments - they are post-deployment validation. While what we want here is to prevent deployments of things that violate rules completely, which CloudFormation Hooks give us.

Reply:
Can we support both options? Feels like there are valid uses for both technologies and I am wondering if the launchpad spec should be open to supporting both.

Reply:
Sure. I think it's viable to support multiple compliance tools.

I would start with CFN Gard + CloudFormation Hooks though, as the first supported one.

Reply:
Config supports preventative controls to block - https://aws.amazon.com/blogs/aws/new-aws-config-rules-now-support-proactive-compliance/